stdlib/public/core/Integers.swift.gyb

```diff
@@ -1199,6 +1199,14 @@ public protocol SignedNumeric : Numeric {
   ///     var x = 21
   ///     x.negate()
   ///     // x == -21
+  ///
+  /// The resulting value must be representable within the value's type. In
+  /// particular, negating a signed, fixed-width integer type's minimum
+  /// results in a value that cannot be represented.
+  ///
+  ///     var y = Int8.min
+  ///     y.negate()
+  ///     // Overflow error
```

Comment:
Is it worth clarifying what options conforming types have when this happens? (Trap, produce a NaN-equivalent, produce the same value, produce an arbitrary value.)

Reply:
Thanks for the comment — correct that this isn't a precondition for all numeric types.

Reply:
That wasn't exactly the question; there's the requirement on callers (the precondition you're talking about), and then there's the requirement on implementers. Both need to be documented here, right?